src/IB/IBFEMethod.cpp

```diff
@@ -980,6 +1048,12 @@ IBFEMethod::forwardEulerStep(const double current_time, const double new_time)
         IBTK_CHKERRQ(ierr);
         d_X_new_vecs[part]->close();
         d_X_half_vecs[part]->close();
+
+        if (d_direct_forcing_kinematics_data[part])
```

Comment:
Do we want to do the normal update followed by the direct-forcing update, or should it be an if statement --- if there are specified kinematics, use those kinematics; otherwise, use the interpolated velocity?

Reply:
I think former is more flexible in the sense that implementation of direct-forcing can skip the position update if they want and just use what IBFEMethod does. But I am fine with the other way too. Let me know.

Reply:
OK with me.